core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/utilities/StringTransformation.java

```diff
@@ -0,0 +1,312 @@
+/**
+ * This software is released under the University of Illinois/Research and Academic Use License. See
+ * the LICENSE file in the root folder for details. Copyright (c) 2016
+ *
+ * Developed by: The Cognitive Computation Group University of Illinois at Urbana-Champaign
+ * http://cogcomp.cs.illinois.edu/
+ */
+package edu.illinois.cs.cogcomp.core.utilities;
+
+import edu.illinois.cs.cogcomp.core.datastructures.IntPair;
+import edu.illinois.cs.cogcomp.core.datastructures.Pair;
+import gnu.trove.map.hash.TIntIntHashMap;
+
+import java.util.*;
+
+/**
+ * Store a list of transformations applied to an original string to produce a new String.
+ * These transformations modify the character offsets of the remaining string (e.g. replacing xml-escaped
+ *     characters with their ascii counterparts, without padding the replacement with whitespace).
+ * This could include deletion of markup, in which case information associated with the markup might be stored.
+ *
+ * @author mssammon
+ */
+public class StringTransformation {
+
+    private final String origText;
+    // tracks whether transformedText field is out of date
```

Comment:
Why are these not in proper JavaDoc format? Rather than the "//", use "/**" to open and "*/" to close, the comment will be added to the javadocs...